File: mobile-engage/src/androidTest/java/com/emarsys/mobileengage/testUtil/ApplicationTestUtils.java
package com.emarsys.mobileengage.testUtil;

import android.app.Application;
import android.content.pm.ApplicationInfo;

import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

public class ApplicationTestUtils {

    public static Application applicationDebug() {
        Application applicationDebug = mock(Application.class);
        ApplicationInfo info = new ApplicationInfo();
        info.flags = ApplicationInfo.FLAG_DEBUGGABLE;
        when(applicationDebug.getApplicationInfo()).thenReturn(info);
        return applicationDebug;
    }

    public static Application applicationRelease() {
        Application applicationRelease = mock(Application.class);
        ApplicationInfo info = new ApplicationInfo();
        info.flags = 0;
        when(applicationRelease.getApplicationInfo()).thenReturn(info);
        return applicationRelease;
    }
}

File: testUtils/src/main/java/com/emarsys/testUtil/ApplicationTestUtils.kt
package com.emarsys.testUtil

import android.app.Application
import android.content.Context
import android.content.pm.ApplicationInfo
import io.mockk.every
import io.mockk.spyk

object ApplicationTestUtils {

    @JvmStatic
    val applicationDebug: Application
        get() = getApplication { flags = ApplicationInfo.FLAG_DEBUGGABLE }

    @JvmStatic
    val applicationRelease: Application
        get() = getApplication { flags = 0 }

    private fun getApplication(init: ApplicationInfo.() -> Unit) =
        (spyk<Context>(InstrumentationRegistry.getTargetContext().applicationContext) as Application).also {
            every { it.applicationInfo } returns ApplicationInfo().apply(init)
        }

}
